### PR TITLE
Add validation to allow submission only when all fields complete

### PR DESCRIPTION
**Pull request by davidkelliott**

### CODE REVIEW

File: app/models/idea.rb [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+  validates :area_of_interest, presence: true, if: :submitted?
+  validates :business_area, presence: true, if: :submitted?
+  validates :it_system, presence: true, if: :submitted?
+  validates :idea, presence: true, if: :submitted?
+  validates :benefits, presence: true, if: :submitted?
+  validates :impact, presence: true, if: :submitted?
+  validates :involvement, presence: true, if: :submitted?
+
+  def submitted?
+    submission_date
```
Comment (davidkelliott) [inline app/models/idea.rb]: I'm a bit confused here, this works but :submission_date doesn't. I used the docs here - https://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless
I understand what the method is doing, but not how, where is submission_date coming from here?  I'm assuming it's some validates magic but cant find it in the docs.

File: spec/requests/ideas_spec.rb [hunk 2]
```diff
@@ -48,14 +48,44 @@
...
+
+    describe "submitting an idea with missing data" do
+      it "should not set the submission date on an idea where fields are missing" do
+        idea = @user.ideas.create!(
+          title: "Submit idea",
+          area_of_interest: 0,
+          business_area: 0
+        )
+        post idea_submit_path(idea)
+        expect(idea.submission_date).to be_nil
```
Comment (bjpirt) [inline spec/requests/ideas_spec.rb]: I think we should also be checking that the form re-renders with errors in it

File: spec/models/idea_spec.rb [hunk 3]
```diff
@@ -2,20 +2,29 @@
...
-  describe "Relations" do
+  describe "relations" do
     it { should belong_to(:user) }
   end
 
-  describe "Validations" do
+  describe "validations" do
     it { should validate_presence_of(:title) }
+    context "if submitted" do
+      before { allow(subject).to receive(:submitted?).and_return(true) }
```
Comment (bjpirt) [inline spec/models/idea_spec.rb]: Might also be worth testing that they don't get run if not submitted

File: app/models/idea.rb (hunk 1 above)
Comment (bjpirt) [reply app/models/idea.rb]: not sure why :submission_date doesn't work out of the box, but in any case line 13 would be better as `submission_date.present?` as it's a boolean

**Pull request merged**